server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -339,7 +338,9 @@ public Void run() {
      * @return a warning value formatted according to RFC 7234
      */
     public static String formatWarning(final String s) {
-        return String.format(Locale.ROOT, WARNING_FORMAT, escapeAndEncode(s), RFC_7231_DATE_TIME.format(ZonedDateTime.now(GMT)));
+        return WARNING_PREFIX + " "
+                + "\"" + escapeAndEncode(s) + "\"" + " "
```

Comment:
Unrelated to your PR: While I am not aware that any RFC specifies a maximum field length for HTTP headers there are practical limits and I wonder whether we should enforce one (in a follow-up) as deprecation messages *might* be long?